androidApp/src/main/java/com/example/tasky/android/common/viewmodel/MainViewModel.kt

```diff
@@ -17,18 +20,17 @@ class MainViewModel(
     val isLoginStateFlow = isLoginFlow.stateIn(scope = viewModelScope, started = SharingStarted.Eagerly, initialValue = false)
 
     init {
-        viewModelScope.launch {
-            konnection
-                .observeHasConnection()
-                .combine(isLoginFlow) { hasConnection, isLogin ->
-                    isLogin && hasConnection
-                }.collect {
-                    if (it) {
-                        viewModelScope.launch {
-                            agendaRepository.syncAgenda()
-                        }
-                    }
-                }
-        }
+        combine(
+            konnection.observeHasConnection(),
+            isLoginFlow,
+        ) { hasConnection, isLogin ->
+            hasConnection && isLogin
+        }.filter {
+            it
+        }.onEach {
+            viewModelScope.launch {
+                agendaRepository.syncAgenda()
+            }
```

Comment:
Is it intentional that you launch a separate coroutine here? You could sync directly in the Flow operator without the launch block. This would also allow the Flow to wait for a sync to finish until it processes the next emission.